platforms/jvm/plugins-java-base/src/main/java/org/gradle/api/plugins/JavaPluginExtension.java

```diff
@@ -289,4 +290,12 @@ public interface JavaPluginExtension {
      * @since 7.1
      */
     boolean getAutoTargetJvmDisabled();
+
+    /**
+     * Configures whether Preview Features should be enabled ({@code --enable-preview} flag).
+     *
+     * @since 8.8
+     */
+    @Incubating
+    Property<Boolean> getEnablePreview();
```

Comment:
I'd say that's the whole point of reducing friction.
See the current boilerplate we suggest in the docs for the users if they want to try out preview features:
https://docs.gradle.org/current/userguide/building_java_projects.html#sec:feature_preview
(and it misses the Javadoc tasks)

Users would still be able to disable the preview per compilation, if they want to. But I would imagine, that this is a much more rare case.

Reply:
To clarify, the intention of the PR is to wire the project level "enable preview" convention to all related java tasks (compile, test, exec, javadoc), specifically to make the trying-out-new-preview-features use case simple.

I will complete the implementation once we agree on the project level extension. Otherwise, I don't see a lot of value in only introducing the task-level switches.